content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,221 +11,289 @@ weight: 10
-{{< /note >}}
+  In other words, if the host mounts anything inside the volume mount, the
+  container will see it mounted there.
 
-See the [CephFS example](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/volumes/cephfs/) for more details.
+  Similarly, if any pod with `Bidirectional` mount propagation to the same
+  volume mounts anything there, the container with `HostToContainer` mount
+  propagation will see it.
 
-### cinder
+  This mode is equal to `rslave` mount propagation as described in the
+  [Linux kernel documentation](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt)
 
-{{< note >}}
-Kubernetes must be configured with the OpenStack cloud provider.
-{{< /note >}}
+* `Bidirectional` - This volume mount behaves the same the `HostToContainer` mount.
+  In addition, all volume mounts created by the container will be propagated
+  back to the host and to all containers of all pods that use the same volume.
 
-The `cinder` volume type is used to mount the OpenStack Cinder volume into your pod.
+  A typical use case for this mode is a pod with a FlexVolume or CSI driver or
+  a pod that needs to mount something on the host using a `hostPath` volume.
 
-#### Cinder volume configuration example
+  This mode is equal to `rshared` mount propagation as described in the
+  [Linux kernel documentation](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt)
+
+  {{< warning >}}
+  `Bidirectional` mount propagation can be dangerous. It can damage
+  the host operating system and therefore it is allowed only in privileged  
+  containers. Familiarity with Linux kernel behavior is strongly recommend  ed.
+  In addition, any volume mounts created by containers in pods must be destroyed
+  (unmounted) by the Containers on termination.
+  {{< /warning >}}
+
+### Mount Propagation Configuration
+Before mount propagation can work properly on some deployments (CoreOS,
+RedHat/Centos, Ubuntu) mount share must be configured correctly in
+Docker as shown below.
+
+Edit your Docker's `systemd` service file. Set `MountFlags` as follows:
+```shell
+MountFlags=shared
+```
+
+Or, remove `MountFlags=slave` if present. Then restart the Docker daemon:
+
+```shell
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+
+## Specifiy subPaths on Volumes {#using-subpath}
+
+Sometimes, it is useful to share one volume for multiple uses in a single pod. The `volumeMounts.subPath` property specifies a sub-path inside the referenced volume instead of its root.
+
+The following example shows how to configure a Pod with a LAMP stack (Linux Apache MySQL PHP)
+using a single, shared volume. This sample `subPath` configuration is not recommended
+for production use.
+
+The PHP application's code and assets map to the volume's `html` folder and
+the MySQL database is stored in the volume's `mysql` folder. For example:
```

Comment:
This also feels like a task more than a concept.

Reply:
I'll move "Using subPath with expanded environment variables" out, but I think it's worth keeping the general discussion of subpaths here, as it's a simple and relevant feature.